opts/hosts.go

```diff
@@ -161,21 +161,53 @@ func ParseTCPAddr(tryAddr string, defaultAddr string) (string, error) {
 	return fmt.Sprintf("tcp://%s%s", net.JoinHostPort(host, port), u.Path), nil
 }
 
-// ValidateExtraHost validates that the specified string is a valid extrahost and returns it.
-// ExtraHost is in the form of name:ip where the ip has to be a valid ip (IPv4 or IPv6).
+// ValidateExtraHost validates that the specified string is a valid extrahost and
+// returns it. ExtraHost is in the form of name:ip or name=ip, where the ip has
+// to be a valid ip (IPv4 or IPv6). The address may be enclosed in square
+// brackets.
 //
-// TODO(thaJeztah): remove client-side validation, and delegate to the API server.
+// For example:
+//
+//	hostname:127.0.0.1
+//	hostname:::1
+//	hostname=::1
+//	hostname:[::1]
```

Comment:
See my other comment; perhaps use `my-hostname` or something similar, to prevent `hostname` being confused for a special keyword.

Reply:
Makes sense - will do!